contracts/src/finance/vesting/implementation.rs

```diff
@@ -0,0 +1,243 @@
+        self.schedules.insert(
+            (to, asset, id),
+            &VestingSchedule {
+                start: vesting_start,
+                end: vesting_end,
+                amount,
+                released: 0,
+            },
+        );
+        self.next_id.insert((to, asset), &(id + 1));
+        Ok(())
+    }
+
+    fn release(
+        &mut self,
+        to: AccountId,
+        asset: Option<AccountId>,
+    ) -> Result<Balance, VestingError> {
+        let next_id = self.next_id.get((to, asset)).unwrap_or(0);
+        let mut tail_id = next_id - 1;
+        let mut current_id = 0;
+        let mut total_amount = 0;
+        while current_id <= tail_id {
+            let (was_swapped_for_tail, amount_released) =
+                self.release_by_vest_id(to, asset, current_id)?;
+            total_amount += amount_released;
+            if was_swapped_for_tail {
+                tail_id = tail_id.saturating_sub(1);
+                continue;
+            }
+            current_id += 1;
+        }
+        Ok(total_amount)
+    }
+
+    fn release_by_vest_id(
+        &mut self,
+        to: AccountId,
+        asset: Option<AccountId>,
+        id: u32,
+    ) -> Result<(bool, Balance), VestingError> {
+        let mut data = match self.schedules.get(&(to, asset, id)) {
+            Some(data) => data,
+            None => return Ok((false, 0)),
+        };
+        let amount_released = data.collect_releasable_rdown();
+        if data.is_overdue() {
+            let leftover = data.amount - data.released;
+            let next_id = self.next_id.get((to, asset)).unwrap(); // data is some => next_id must exist and be > 0
+            let tail_id = next_id - 1;
+            let tail = self
+                .schedules
+                .get(&(to, asset, tail_id))
+                .ok_or(VestingError::InvalidScheduleKey)?;
+            self.schedules.remove(&(to, asset, tail_id));
+            if tail_id != id {
+                self.schedules.insert((to, asset, id), &tail);
+            }
+            self.next_id.insert((to, asset), &(tail_id));
+            return Ok((true, amount_released + leftover));
+        }
+        self.schedules.insert((to, asset, id), &data);
+        Ok((false, amount_released))
+    }
+
+    fn get_schedule_by_id(
+        &self,
+        to: AccountId,
+        asset: Option<AccountId>,
+        id: u32,
+    ) -> Option<VestingSchedule> {
+        self.schedules.get(&(to, asset, id))
+    }
+}
+
+pub trait VestingDefaultImpl: VestingInternal + Sized {
+    fn create_vest_default_impl(
+        &mut self,
+        to: AccountId,
+        asset: Option<AccountId>,
+        amount: Balance,
+        vesting_start: Timestamp,
+        vesting_end: Timestamp,
+    ) -> Result<(), VestingError> {
+        self._create_vest(to, asset, amount, vesting_start, vesting_end)
+    }
+
+    fn release_default_impl(&mut self, asset: Option<AccountId>) -> Result<(), VestingError> {
```

Comment:
add "to: AccountId" argument so the owner of the vest can release to the specified 'to' account.